vue-custom-bootstrap/src/views/Assessment.vue

```diff
@@ -0,0 +1,93 @@
+
+<template>
+  <div class="assessment">
+    <FormSection title="Basic Info" @submit="submitForm(formValues)">
```

Comment:
This `<component>` component is a magical component that vue has built in. It allows us to dynamically transform it into any other component by specifying the 'is' prop. That allows us to loop through the "formData" values and use whatever component we want (text input, radio, checkbox).

Reply:
Yes.

Reply:
<component
        v-for="(item, index) in formData"
        :key="index"
        :is="item.**_fieldType_**"
        v-model="formValues[item.name]"
        v-bind="item"
      />

later , if we need add new UI type, we have to 
(1) create new component in front end 
(2) build SMIS and release 
?

Reply:
That depends on how the database is structured. It raises a couple of other questions too:
* do we want control of how the UI layout works, such as two fields side by side?
* can we do any front end validation, or is it all in a database as well?

Overall, is it easier to redeploy the front end for these types of changes than it is to update the back end data source?